test/main.js

```diff
@@ -173,32 +175,96 @@ const runner = async () => {
         if (driver) driver.finish()
       }
     })
-    console.log('Finished test')
   })
 
-  console.log('finished')
+  console.log(chalk.green('Tests finished'))
   process.exit(totalFailures > 0 ? 1 : 0)
 }
 
-const server = exec('npm run travis')
-const killServer = () => {
-  console.log('Killing server')
-  if (!server.killed) {
-    server.kill()
-    console.log('Kill signal sent')
-  } else {
-    console.log('Kill signal already sent')
+const killMockServer = (dockerContainerId) => {
+  console.log(chalk.grey('Killing mock server'))
+  exec(`docker stop ${dockerContainerId} -t0`, (error) => {
+    if (error) {
+      console.log(chalk.yellow('Error killing mock server:'), error)
+    } else {
+      console.log(chalk.green('Mock server killed'))
+    }
+  })
+}
+
+console.log(chalk.bold.green('Starting mock server'))
+
+const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms))
+
+const pingMockServer = () => {
+  const options = {
+    hostname: 'localhost',
+    port: 8080,
+    path: '/api/ping',
+    method: 'GET',
+    rejectUnauthorized: false,
   }
+
+  return new Promise((resolve, reject) => {
+    https
+      .request(options, (res) => {
+        res.on('data', (data) => resolve(data))
+      })
+      .on('error', reject)
+      .end()
+  })
 }
 
-server.stdout.on('data', (data) => {
-  if (data.includes('Available on:')) {
-    runner()
+/**
+ * There's always an amount of delay time for the mock server to be alive completely,
+ * after the `docker run ...` command exits with the container ID.
+ * Without this waiting method, we can see several first tests to be failed,
+ * which is clearly a false positive.
+ * Particularly the sleep(1000) is to defer the ping action to be run every 1 second,
+ * instead of every process tick, which is unnecessarily wasteful.
+ */
+const waitForMockServer = async () => {
+  let isMockServerRunning = false
+
+  while (!isMockServerRunning) {
```

Comment:
just a quick question. What if some error occurs and mock server never starts. Shuldn't we have a 30s/60s timeout to error everything?

Reply:
Good catch, I'll update to handle this bit better